packages/world/ts/node/render-solidity/renderSystemLibrary.ts

```diff
@@ -0,0 +1,260 @@
+import {
+  renderArguments,
+  renderList,
+  renderedSolidityHeader,
+  renderImports,
+  ContractInterfaceFunction,
+} from "@latticexyz/common/codegen";
+import { RenderSystemLibraryOptions } from "./types";
+import { ContractInterfaceError } from "@latticexyz/common/codegen";
+
+export function renderSystemLibrary(options: RenderSystemLibraryOptions) {
+  const {
+    imports: systemImports,
+    libraryName,
+    systemLabel,
+    resourceId,
+    namespace,
+    functions: systemFunctions,
+    errors: systemErrors,
+    worldImportPath,
+    storeImportPath,
+  } = options;
+
+  // Add required imports, if they are already included they will get removed in renderImports
+  const imports = [
+    ...systemImports,
+    {
+      symbol: "revertWithBytes",
+      path: `${worldImportPath}/revertWithBytes.sol`,
+    },
+    {
+      symbol: "IWorldCall",
+      path: `${worldImportPath}/IWorldKernel.sol`,
+    },
+    {
+      symbol: "SystemCall",
+      path: `${worldImportPath}/SystemCall.sol`,
+    },
+    {
+      symbol: "Systems",
+      path: `${worldImportPath}/codegen/tables/Systems.sol`,
+    },
+    {
+      symbol: "ResourceId",
+      path: `${storeImportPath}/ResourceId.sol`,
+    },
+    {
+      symbol: "StoreSwitch",
+      path: `${storeImportPath}/StoreSwitch.sol`,
+    },
+  ];
+
+  const callingFromRootSystemErrorName = `${libraryName}_CallingFromRootSystem`;
+  const errors = [{ name: callingFromRootSystemErrorName, parameters: [] }, ...systemErrors];
+
+  const camelCaseSystemLabel = systemLabel.charAt(0).toLowerCase() + systemLabel.slice(1);
+  const userTypeName = `${systemLabel}Type`;
+
+  // Remove view functions for root systems
```

Comment:
should we add some more details to this comment for why we have to remove view function for root systems here? (similar to the comment in `SystemCall.staticcall`)

Reply:
this was not necessary, will add those functions back to the root system libs